Lib/test/test_capi/test_opt.py

```diff
@@ -2362,6 +2364,23 @@ def testfunc(n):
         self.assertNotIn("_GUARD_TOS_INT", uops)
         self.assertNotIn("_GUARD_NOS_INT", uops)
 
+    def test_store_fast_pop_top_specialize_unicode(self):
+        def random_str(n):
+            return ''.join(random.choice(string.ascii_uppercase + string.digits) for _ in range(n))
+        def testfunc(n):
+            y = random_str(32)
+            for _ in range(n):
+                x = y + y # _POP_TOP
+                x = None # _POP_TOP_NOP
+
+        testfunc(TIER2_THRESHOLD)
+
+        ex = get_first_executor(testfunc)
+        self.assertIsNotNone(ex)
+        uops = get_opnames(ex)
+
+        self.assertIn("_POP_TOP_NOP", uops)
```

Comment:
Maybe also add
```python
self.assertIn("_BINARY_OP_ADD_UNICODE", uops)
```
to make sure this is the op that's actually generated?

Should we also test for `_POP_TOP_UNICODE`?

Reply:
Yup, do you think it makes sense to test `_POP_TOP_UNICODE` separately? That is, the case when it is not optimized to `_POP_TOP_NOP`? (not sure how difficult it'll be to come up with a test case though)